src/client/auth/aws.rs

```diff
@@ -171,168 +226,183 @@ pub(crate) struct AwsCredential {
     expiration: Option<crate::bson::DateTime>,
 }
 
-fn non_empty(s: Option<String>) -> Option<String> {
-    match s {
-        None => None,
-        Some(s) if s.is_empty() => None,
-        Some(s) => Some(s),
-    }
-}
+// fn non_empty(s: Option<String>) -> Option<String> {
+//     match s {
+//         None => None,
+//         Some(s) if s.is_empty() => None,
+//         Some(s) => Some(s),
+//     }
+// }
 
 impl AwsCredential {
-    /// Derives the credentials for an authentication attempt given the set of credentials the user
-    /// passed in.
-    pub(crate) async fn get(credential: &Credential, http_client: &HttpClient) -> Result<Self> {
-        let access_key = credential
-            .username
-            .clone()
-            .or_else(|| non_empty(std::env::var("AWS_ACCESS_KEY_ID").ok()));
-        let secret_key = credential
-            .password
-            .clone()
-            .or_else(|| non_empty(std::env::var("AWS_SECRET_ACCESS_KEY").ok()));
-        let session_token = credential
-            .mechanism_properties
-            .as_ref()
-            .and_then(|d| d.get_str("AWS_SESSION_TOKEN").ok())
-            .map(|s| s.to_string())
-            .or_else(|| non_empty(std::env::var("AWS_SESSION_TOKEN").ok()));
-
-        let found_access_key = access_key.is_some();
-        let found_secret_key = secret_key.is_some();
-
-        // If we have an access key and secret key, we can continue with the credentials we've
-        // found.
-        if let (Some(access_key), Some(secret_key)) = (access_key, secret_key) {
-            return Ok(Self {
-                access_key_id: access_key,
-                secret_access_key: secret_key,
-                session_token,
-                expiration: None,
-            });
+    // /// Derives the credentials for an authentication attempt given the set of credentials the
+    // user /// passed in.
+    // pub(crate) async fn get(credential: &Credential, http_client: &HttpClient) -> Result<Self> {
```

Comment:
Rather than comment out all of these methods, I suggest just adding the `#[allow(dead_code)]` annotation with a note that they'll be addressed when we decide on the feature flag situation - that should make this diff a bit more readable